### PR TITLE
Allow setting a custom state-badge icon color

### PR DESCRIPTION
**Pull request by NovapaX**
This will allow setting a 'icon_color' attribute (like the 'icon' attribute we already have)
It will simple set the value as the css color value.
So you can do `icon_color: black`, `icon_color: rgba()` or any other valid css.

Setting an `icon_color` will skip any other color or brightness modification to the icon.

P.S. this is mainly to help people who use custom UI and are forced to do all kind of wierd hue/saturation calculation to get thier icons colored the right way.
I know the state is not the perfect place for this. But it's neatly coupled with `icon` now so they can move together once there's a better place.

### CODE REVIEW

File: src/components/entity/state-badge.html [hunk 1]
```diff
@@ -75,6 +75,8 @@
     if (newVal.attributes.entity_picture) {
       hostStyle.backgroundImage = 'url(' + newVal.attributes.entity_picture + ')';
       iconStyle.display = 'none';
+    } else if (newVal.attributes.icon_color) {
+      iconStyle.color = `${newVal.attributes.icon_color}`;
```
Comment (c727) [inline src/components/entity/state-badge.html]: no need for the template

Comment (NovapaX) [reply src/components/entity/state-badge.html]: my goal was to cast/force it to string, so we don't break stuff.

**Comment by balloob** [comment]
I don't think that we should merge this.

We've had a whole discussion on the abuse of state attributes for the sake of frontend. We decided we want less of them, not more. Icon and entity picture are the only two that are grandfathered in, we definitely should not add any more.

Once something like [my frontend proposal](https://github.com/home-assistant/architecture/issues/14) has been implemented, things like this are going to be a lot easier, as we no longer expect people to pollute the state machine with frontend related info.

**Comment by andrey-git** [comment]
Note that the frontend used to set the icon color according to `rgb_color` (array) attribute before #1021

**Comment by balloob** [comment]
The intention of that logic was supposed to be just for lights.

What are use cases where people want to color their icon outside of lights?

**Comment by NovapaX** [comment]
I can think of a few reasons (some more valid than others)
- Theming: bad reason
- Additional visual identification or differentiation for specific entities
- Visual notifications (e.g. red icon when sensor value above/below certain threshold, so colorizing based on state, like we do with lights)

That first case should be covered by theming. When we have a new frontend-config stuff going we could allow to set theming variables for specific parts of the interface (so all icons in a card could have a different active-color.

Second case should be in frontend configuration for that specific entity.

Last case seems kind of valid. Most of the times this is done with the custom_ui component that allows templated attributes. https://github.com/andrey-git/home-assistant-custom-ui/blob/master/docs/features.md#template-attributes 
We could add attributes like 'warning_level' and hardcoded colors or adding even more variables. But then we'd have covered a single use case (warning thresholds). What if I want to color a rest-sensor with trash-can icon green/black/blue based on the type of garbage I need to put on the side of the road today)
Instead we could just simply allow setting an icon_ color by name (or any other valid css color) and cover a lot of the possible use-cases.

**Comment by arsaboo** [comment]
There is definitely a need for something like this (may not be this specific PR) that allows users to use visual cues in the front end. For example, I would love the icon to turn red when the stock market is firm and green when it is up. The use of such visual cues also allows us to add more information to the front end, which is particularly useful for mobile devices. The change to hs_color has messed up the use of custom_ui ability to change colors.

**Comment by balloob** [comment]
So you want to have a visual cue so when you glance over states, you can see which one requires attention.

That's a use case that should not be solved by adding a new generic attribute that people should add to the state machine to help STYLE the frontend. States are for states, not UI directives.

We are also definitely not going to add something as a temporary solution. People are already throwing a tantrum because we removed a feature that we did not know we had. Imagine removing a feature that is in the docs.

**Comment by NovapaX** [comment]
So custom UI is the answer here. 
Let's just work on making that more accessible for less experienced users.
note: Of course, if one wants, the `hs_color` and `brightness` can still be 'abused' for coloring the icon.

**Pull request closed without merging**